src/futures/update_stream.rs

```diff
@@ -0,0 +1,230 @@
+
+    //update calls from the validator
+    let updater = async move {
+      while let EventResult::Continue = self.update_call().await {}
+      EventResult::Shutdown
+    }
+    .fuse();
+
+    //publishing resetter
+    let schedule_timer = Fuse::terminated();
+    //commit listener
+    let commiter = UpdateStream::toggle_on_reactor(commit_flag.clone()).fuse();
+
+    pin_mut!(scheduler, schedule_timer, updater, commiter);
+
+    //Schedule a publishing timer, when the timer yields, toggle on a flag and spawn a task that waits for the flag to turn off.
+    //Schedule a commit reactor, when the flag turns on, force-reset the publishing timer.
+    loop {
+      select! {
+        // timer
+        () = scheduler=>{
+          //Publishing time, schedule reset.
+            schedule_timer.set(UpdateStream::toggle_off_reactor(publishing_flag.clone()).fuse());
+        },
+        () = schedule_timer =>{
+          scheduler.set( PublishSchedulerFuture::schedule_publishing(publishing_flag.clone(), time).fuse());
+        },
+        () = commiter =>{
+          //reset publisher timer
+            scheduler.set( PublishSchedulerFuture::schedule_publishing(publishing_flag.clone(), time).fuse());
+            schedule_timer.set(Fuse::terminated());
+          //reschedule the reactor;
+            commit_flag.clone().store(false,Ordering::Release);
+            commiter.set(UpdateStream::toggle_on_reactor(commit_flag.clone()).fuse());
+          //force publish TODO?
+
+        },
+        _ = updater => break,
+        complete =>{}
+      }
+    }
+  }
+}
+
+impl UpdateStream {
+  async fn update_call(&mut self) -> EventResult {
+    if self.publishing_flag.load(Ordering::Acquire) {
+      match self.node.try_publish() {
+        Ok(EventPublishResult::Published) => {
+          trace!("Resetting publishing flag");
+          #[cfg(feature = "test-futures")]
+          println!("Resetting publishing flag");
+          self.publishing_flag.store(false, Ordering::Release);
+        }
+        Ok(..) => {}
+        Err(e) => {
+          warn!(
+            "Publishing Error {}. Consensus event handler is stopping.",
+            e
+          );
+          return EventResult::Shutdown;
+        }
+      }
+    }
+
+    let call = self.updates.recv_timeout(Duration::from_millis(0));
+    match call {
+      Ok(update) => {
+        trace!("Incoming update {:?}", update);
+        match self.node.handle_update(update) {
+          Ok(true) => EventResult::Continue,
+          Ok(false) => EventResult::Shutdown,
+          Err(error) => {
+            error!("Update Error: {}", error);
+            EventResult::Continue
+          }
+        }
+      }
+      Err(RecvTimeoutError::Disconnected) => {
+        error!("Disconnected from validator");
+        EventResult::Shutdown
+      }
+      Err(RecvTimeoutError::Timeout) => {
+        sleep(Duration::from_millis(10)).await;
+        EventResult::Continue
+      }
+    }
```

Comment:
```suggestion
    match self.updates.try_recv() {
      Ok(update) => {
        trace!("Incoming update {:?}", update);
        match self.node.handle_update(update) {
          Ok(true) => EventResult::Continue,
          Ok(false) => EventResult::Shutdown,
          Err(error) => {
            error!("Update Error: {}", error);
            EventResult::Continue
          }
        }
      }
      Err(std::sync::mpsc::TryRecvError::Disconnected) => {
        error!("Disconnected from validator");
        EventResult::Shutdown
      }
      Err(std::sync::mpsc::TryRecvError::Empty) => {
        sleep(Duration::from_millis(10)).await;
        EventResult::Continue
      }
    }
```
Instead of using the blocking call `recv_timeout` with a timeout of `0`, we can use the `try_recv` method which is intended to be non-blocking.